### PR TITLE
feat(react): show search suggestions

### PR DESCRIPTION
**Pull request by gabe-lyons**
Search suggestions are fetched from elastic by querying for `*` and then randomizing the order of the result

image

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/home/HomePageHeader.tsx [hunk 1]
```diff
@@ -57,8 +110,37 @@ export const HomePageHeader = () => {
...
+    const suggestionsToShow = useMemo(() => {
+        let result: string[] = [];
+        if (!suggestionsLoading) {
+            [EntityType.Dashboard, EntityType.Chart, EntityType.Dataset].forEach((type) => {
+                const suggestionsToShowForEntity = getSuggestionFieldsFromResult(
+                    allSearchResultsByType[type]?.data,
+                ).sort(sortRandom);
+                const suggestionToAddToFront = suggestionsToShowForEntity?.pop();
+                result = [...result, ...suggestionsToShowForEntity];
+                if (suggestionToAddToFront) {
```
Comment (dexter-mh-lee) [inline datahub-web-react/src/app/home/HomePageHeader.tsx:134]: What is this logic doing?

**Review by dexter-mh-lee — COMMENTED** [review]
Overall logic LGTM. Few questions on latency. How does the load time look? Does the whole page load after all the search requests finish?

File: datahub-web-react/src/app/home/HomePageHeader.tsx (hunk 1 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/home/HomePageHeader.tsx:134]: This interleaves the results of different types of queries so we have a variety of entity types suggested

**Comment by gabe-lyons** [comment]
@dexter-mh-lee all these queries are issued client side, so nothing is shown here until the response comes back. On my machine, its near instantaneous though.

File: datahub-web-react/src/app/home/HomePageHeader.tsx (hunk 1 above)
Comment (dexter-mh-lee) [reply datahub-web-react/src/app/home/HomePageHeader.tsx:134]: Got it

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]

**Pull request merged**